search/terms_aggregation.go

```diff
@@ -0,0 +1,201 @@
+package search
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// TermsAggregation dynamically creates a bucket for each unique term of a field.
+// An empty TermsAggregation will have two issues with execution:
+//
+//   - the target Field must be non-null and non-empty.
+//   - a Size of 0 will return no buckets
+//
+// For more details see https://opensearch.org/docs/latest/opensearch/bucket-agg/
+type TermsAggregation struct {
+	// Field to be bucketed
+	Field string
+
+	// Size of the number of buckets to be returned. Negative sizes will be omitted
+	Size int
+
+	// Order list of [Order]s to sort the aggregation buckets. Default order is _count: desc
+	Order []Order
+
+	// Aggregations sub aggregations for each bucket. Mapped by string label to sub aggregation
+	Aggregations map[string]Aggregation
+}
+
+// NewTermsAggregation instantiates a TermsAggregation targeting the provided field
+// and sets the Size to -1 to be omitted for the default value.
+func NewTermsAggregation(field string) *TermsAggregation {
+	return &TermsAggregation{
+		Field:        field,
+		Size:         -1,
+		Aggregations: make(map[string]Aggregation),
+	}
+}
+
+// WithSize for the number of buckets to be returned
+func (t *TermsAggregation) WithSize(size int) *TermsAggregation {
```

Comment:
Maybe call out the valid range for `size` and make this a no-op if passed an invalid value.  I wouldn't change the signature to return (*TermsAggregation, error)` though.

Reply:
It's less an "invalid" value as it is an ignored value to leverage opensearch's default value. I'm not sure what the usecase would be for someone to set a size, and the later reuse the same struct and want to remove the size from the aggregation. But it's not impossible. So idk about no-oping.

Reply:
What would `.WithSize(-9000)` do?  I brought it up because I see we're defaulting to -1.

Definitely not a big deal, just wondering if we can make the _Pit of Success_ bigger.

Reply:
Anything negative is ignored in the `ToOpenSearchJSON` method. The unit test covering it uses `WithSize(-5)`.